gorilla/example_test.go

```diff
@@ -0,0 +1,74 @@
+package gorilla_test
+
+import (
+	"net/http"
+	"log"
+	"fmt"
+	"time"
+
+    "github.com/gorilla/mux"
+	"github.com/opentracing/opentracing-go"
+	"github.com/uber/jaeger-client-go/config"
+    "github.com/opentracing-contrib/go-gorilla/gorilla"
+)
+
+func NewServeMux(tracer opentracing.Tracer) *TracedServeMux {
```

Comment:
isn't there a better way to implement this with Gorilla than wrapping the router? You are only overriding a single function Handle, but there are a lot more ways of attaching handlers to Gorilla router, so all those ways are not going to be traced, which creates a poor / misleading experience to the user of this package.

Reply:
Yes, I got rid of this fat struct,  given a simple example as posibble. @yurishkuro